jfix-stdlib-socket/src/main/java/ru/fix/stdlib/socket/proxy/ProxySocket.java

```diff
@@ -0,0 +1,91 @@
+    private int destinationPort;
+    private int sourcePort;
+
+    private ServerSocket sourceServerSocket;
+    private AtomicBoolean isShutdown = new AtomicBoolean();
+
+    private ExecutorService executorService = Executors.newFixedThreadPool(25);
+
+    public ProxySocket(String destinationHost, int destinationPort, int sourcePort) throws IOException {
+        this.destinationHost = destinationHost;
+        this.destinationPort = destinationPort;
+        this.sourcePort = sourcePort;
+
+        start();
+    }
+
+    private void start() throws IOException {
+        sourceServerSocket = new ServerSocket(sourcePort);
+        executorService.submit(() -> {
+                    final byte[] request = new byte[1024];
+                    final byte[] reply = new byte[4096];
+
+                    while (!isShutdown.get()) {
+                        try (Socket sourceSocket = sourceServerSocket.accept();
+                             final InputStream streamFromClient = sourceSocket.getInputStream();
+                             final OutputStream streamToClient = sourceSocket.getOutputStream();
+                             Socket destinationSocket = new Socket(destinationHost, destinationPort)) {
+                            executorService.submit(() -> {
+                                try (OutputStream streamToServer = destinationSocket.getOutputStream()) {
+                                    int bytesRead;
+                                    while (!isShutdown.get() && (bytesRead = streamFromClient.read(request)) != -1) {
+                                        streamToServer.write(request, 0, bytesRead);
+                                        streamToServer.flush();
+                                    }
+                                } catch (IOException e) {
+                                    log.error("Failed to flush to dest {}", e);
+                                }
+                            });
+
+                            try (InputStream streamFromServer = destinationSocket.getInputStream()) {
+                                int bytesRead;
+                                while (!isShutdown.get() && (bytesRead = streamFromServer.read(reply)) != -1) {
+                                    streamToClient.write(reply, 0, bytesRead);
+                                    streamToClient.flush();
+                                }
+                            } catch (IOException e) {
+                                log.error("Failed to flush to client", e);
+                            }
+                        } catch (IOException e) {
+                            log.error("Failed to open socket", e);
+                        }
+                    }
+                }
+        );
+    }
+
+    public int getPort() {
+        return sourceServerSocket.getLocalPort();
+    }
+
+    @Override
+    public void close() throws Exception {
+        isShutdown.set(true);
+
+        executorService.shutdown();
```

Comment:
probably we should close socket as a first step, then wait for termination and if fail to do that during 5 seconds write error message and force process with shutdownNow